java/code/src/com/redhat/rhn/frontend/action/rhnpackage/LockPackageAction.java

```diff
@@ -61,8 +64,18 @@
 
     @Override
     protected DataResult getDataResult(Server server) {
-        DataResult result = PackageManager.systemTotalPackages(server.getId(), null);
-        return result;
+        Optional<MinionServer> minion = MinionServerFactory.lookupById(server.getId());
+        // Check if this server is a minion
+        boolean isMinion = minion.isPresent();
+        LOG.debug(server.getId() + "is a minion system? " + isMinion);
+        // Check if this is a SUSE system (for minions only)
+        boolean isSUSEMinion = isMinion && minion.get().getOsFamily().equals("Suse");
+        LOG.debug(server.getId() + "is a SUSE system? " + isMinion);
```

Comment:
```suggestion
        LOG.debug(server.getId() + "is a SUSE system? " + isSUSEMinion);
```
I suppose it's a typo